accepted/2024/[TS]PMUI-Readme-rendering.md

```diff
@@ -0,0 +1,102 @@
+MarkdownPreviewControl = markdownPreview.VisualElement
+```
+#### Locating the ReadMe
+For packages that are already on the disk we will retrieve the ReadMe from the root folder of the package. 
+
+For remote packages we will download the file and pull the md from the stream response using the [ZipArchive](https://learn.microsoft.com/en-us/dotnet/api/system.io.compression.ziparchive?view=net-8.0) class.
+
+If the feed we're calling allows for [HTTP range requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests) we can implement something similar to [Seekable HTTP Range Stream](https://codereview.stackexchange.com/questions/70679/seekable-http-range-stream) to reduce the amount of data we pull. 
+<!-- Explain the proposal in sufficient detail with implementation details, interaction models, and clarification of corner cases. -->
+
+## Drawbacks
+
+This control is not currently using Daytona. There are plans to upgrade it, [Experience 1870733](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1870733), but they have been delayed. 
+
+It's also currently marked as obsolete since the interface has not been finalized. So when an upgrade is made we may have to change how we use the control.
+
+Currently WebView2 controls always render ontop of other controls in the view. [Secnario 25254665](https://microsoft.visualstudio.com/Edge/_workitems/edit/25254665). PM UI needs to be updated to ensure items don't scroll off screen.
+
+## Rationale and alternatives
+By using an existing control we maintain consistency throughout the IDE and can rely on the owner to fix any bugs with the control.
+<!-- Why is this the best design compared to other designs? -->
+<!-- What other designs have been considered and why weren't they chosen? -->
+<!-- What is the impact of not doing this? -->
+
+## Prior Art
+The IMarkdownPreview is currently being used when creating a new pull request inside of Visual Studio.  
+![Alt text](../../meta/resources/ReadMePMUI/PullRequestExperience.png) 
+<!-- What prior art, both good and bad are related to this proposal? -->
+<!-- Do other features exist in other ecosystems and what experience have their community had? -->
+<!-- What lessons from other communities can we learn from? -->
+<!-- Are there any resources that are relevant to this proposal? -->
+
+## Unresolved Questions
+1. Will we show readme in the updates tab?
+1. What do we show if the package has a ReadMe.txt instead of md?
+    * Current suggestion is to focus on ReadMe.md. Once that's complete see how ReadMe.txt would appear when rendered. 
+    * How many packages have a ReadMe.txt vs ReadMe.md?
+1. What do we show if there is no ReadMe defined?
+    - [ ] Hide the tab from the the details pane
+    - [ ] Display the tab with a message saying there is no ReadMe defined.
+1. Where do we get the ReadMe from when it's not on the disk?
+    - Does the current API return the ReadMe information when searching for a package? 
```

Comment:
Putting my feedback from offline here: 

IMO, we should honor a couple of design principles: 

- No explorative downloads. So no guessing whether there's a readme. The server should tell us if it's there. 
- Some way to create get or create a url to get the readme file. Matches license/icon and avoids attempting partial downloads which raise a bunch of questions.
- 
We need this to be a documented behavior so that other servers can implement it.